src/main/java/org/crac/Context.java

```diff
@@ -46,8 +46,12 @@ public abstract void beforeCheckpoint(Context<? extends Resource> context)
     public abstract void afterRestore(Context<? extends Resource> context)
             throws RestoreException;
 
-    /**
-     * Registers a {@code Resource} with this {@code Context}.
+     /**
+     * Registers a {@code Resource} with this {@code Context}
+      *
+     * <p> Context maintains a weak reference to registered Resource, thus client code must maintain a strong reference
+      * to the Resource to guarantee `beforeCheckpoint` and `afterRestore` methods execution.
```

Comment:
First of all, thank you for the PR!

Regarding the change, should it really be in the description of the abstract `register` method? For example, there can be an another Context implementaion that in contrast maintains strong links (`A {@code Context} implementor is encouraged to respect properties of the global {@code Context}`, but that is not mandatory). But I support any improvements for clarity. Does it make sense to make a reference for the global context properties (https://github.com/CRaC/org.crac/blob/master/src/main/java/org/crac/package-info.java#L68)(https://javadoc.io/doc/io.github.crac/org-crac/latest/index.html, Global Context Properties)) somehow?

Reply:
Hi Anton, I think you are right, documenting implementation details for an abstract method could look confusing. Let me think what would be the best avenue for this and I will get back to you. Thanks for your feedback 👍